cookbook/bundles/tests.rst

```diff
@@ -0,0 +1,311 @@
+=====================================
+
+The Standard Edition's front controller looks like this::
+
+    // web/app.php
+    use Symfony\Component\HttpFoundation\Request;
+
+    require_once __DIR__.'/../vendor/autoload.php';
+    require_once __DIR__.'/../app/AppKernel.php';
+
+    // Create the request from PHP variables super globals such as $_GET, $_POST, $_SERVER, etc
+    $request = Request::createFromGlobals();
+
+    // The application handles the Request
+    $kernel = new AppKernel('prod', false);
+    $response = $kernel->handle($request);
+
+    // Dump the Response's headers and print the Response's content
+    $response->send();
+
+In this example, Symfony is used as an application that mimicks the HTTP protocol,
+which means that it is possible to write transversal tests (System Tests) by building
+"manually" a ``Request``, giving it to our application and then check the returned ``Response``.
+
+This article provides examples on how to write such tests with third party bundles.
+
+Creating a Minimal AppKernel
+----------------------------
+
+Third party bundles cannot be ran standalone, they need an application.
+It is possible to create a minimalistic ``AppKernel`` inside the bundle for testing
+and showcase purposes, instead of:
+
+# Creating a separate application;
+# Install the bundle in it;
+# Then check how things turn out.
+
+If the third party bundle is named ``AcmeMyBundle``, the ``AppKernel`` could look like this::
+
+    // Tests/app/AppKernel.php
+    namespace Acme\MyBundle\Tests;
+
+    use Symfony\Component\HttpKernel\Kernel;
+    use Symfony\Component\Config\Loader\LoaderInterface;
+
+    class AppKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            return array(
+                new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+                new Acme\MyBundle\AcmeMyBundle(),
+            );
+        }
+
+        public function registerContainerConfiguration(LoaderInterface $loader)
+        {
+            $loader->load(__DIR__.'/config.yml');
+        }
+    }
+
+.. note::
+
+    ``AppKernel`` adds a direct dependency on Symfony's FrameworkBundle, which can be
+    installed by running the following command: ``composer require --dev symfony/framework-bundle``.
+
+In its minimum state ``AppKernel`` requires only one configuration parameter, ``secret``:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # Tests/app/config.yml
+        framework:
+            secret: Th1s1sS3cr3t!
+
+Symfony applications generate cache and logs directories, which can be ignored by
+adding those lines to ``.gitignore``:
+
+.. code-block::
+
+    # .gitignore
+    /Tests/app/cache
+    /Tests/app/logs
+    /vendor
+    /composer.lock
+
+Browsing Pages
+--------------
+
+If the third party bundle provides a controller, it can be helpful to create a
+front controller similar to the one that can be found in Symfony's Standard Edition::
```

Comment:
If the point is simply to make it convenient to "try" your controllers, then I'd much rather a third-party bundle author install their bundle into a real project to try it. If we're doing this to functionally test the bundle, we should just instantiate the kernel directly as you've shown below.